components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java

```diff
@@ -337,28 +349,30 @@ public Object toHttpResponse(HttpServerExchange httpExchange, Message message) t
         Exception exception = message.getExchange().getException();
 
         if (exception != null) {
-            if (isTransferException()) {
-                // we failed due an exception, and transfer it as java serialized object
-                ByteArrayOutputStream bos = new ByteArrayOutputStream();
-                ObjectOutputStream oos = new ObjectOutputStream(bos);
-                oos.writeObject(exception);
-                oos.flush();
-                IOHelper.close(oos, bos);
-
-                // the body should be the serialized java object of the exception
-                body = ByteBuffer.wrap(bos.toByteArray());
-                // force content type to be serialized java object
-                message.setHeader(Exchange.CONTENT_TYPE, "application/x-java-serialized-object");
-            } else {
-                // we failed due an exception so print it as plain text
-                StringWriter sw = new StringWriter();
-                PrintWriter pw = new PrintWriter(sw);
-                exception.printStackTrace(pw);
-
-                // the body should then be the stacktrace
-                body = ByteBuffer.wrap(sw.toString().getBytes());
-                // force content type to be text/plain as that is what the stacktrace is
-                message.setHeader(Exchange.CONTENT_TYPE, "text/plain");
+            if (!isMuteException()) {
```

Comment:
Put this on a single line exception != null && !isMute ...

Reply:
I put this on a single line. However, with muteException enabled the exchange will not be marked as failure handled. I'm wondering whether I should include a second condition "exception != null && isMute..." that does exactly this.

Reply:
Ah sorry my bad then, yeah you are welcome to make it so its understandable and maintainable. Or maybe have a if (muteException) else { ... } block etc

Reply:
In commits 2b8ab50 and 95a7150 I put the nested conditions into single line statement and added a second condition to mark the failure as handled since the exception is actively muted.